src/posix/http_client_curl.cc

```diff
@@ -41,12 +44,15 @@ std::string RunCurl(const std::string& cmd) {
   FILE* pipe = ::popen(cmd.c_str(), "r");
   assert(pipe);
   char s[8 * 1024];
-  ::fgets(s, sizeof(s) / sizeof(s[0]), pipe);
+  std::string result;
+  while (nullptr != ::fgets(s, sizeof(s) / sizeof(s[0]), pipe)) {
+    result += s;
```

Comment:
I’d expect from this code to avoid any modification of received content at all, exceptionally for binary data.

Don’t get how is it connected with magic number 3 :)

> On Mar 15, 2015, at 20:40, Dima <notifications@github.com> wrote:
> 
> In src/posix/http_client_curl.cc <https://github.com/KnowSheet/Alohalytics/pull/27#discussion_r26449930>:
> 
> > @@ -41,12 +44,15 @@ std::string RunCurl(const std::string& cmd) {
> >    FILE* pipe = ::popen(cmd.c_str(), "r");
> >    assert(pipe);
> >    char s[8 * 1024];
> > -  ::fgets(s, sizeof(s) / sizeof(s[0]), pipe);
> > +  std::string result;
> > +  while (nullptr != ::fgets(s, sizeof(s) / sizeof(s[0]), pipe)) {
> > +    result += s;
> I'd expect both lines to either have or not have a '\n' at the end.
> 
> Otherwise, the idea of taking a - 3 doesn't look solid at all.
> 
> —
> Reply to this email directly or view it on GitHub <https://github.com/KnowSheet/Alohalytics/pull/27/files#r26449930>.
>